diff -u -r boost_1_47_0/boost/random/detail/integer_log2.hpp boost_1_47_0_tl/boost/random/detail/integer_log2.hpp
--- boost_1_47_0/boost/random/detail/integer_log2.hpp	2011-07-03 06:26:19.000000000 +1000
+++ boost_1_47_0_tl/boost/random/detail/integer_log2.hpp	2011-08-09 09:53:19.000000000 +1000
@@ -28,8 +28,8 @@
 #define BOOST_RANDOM_DETAIL_CONSTEXPR constexpr
 #elif defined(BOOST_MSVC)
 #define BOOST_RANDOM_DETAIL_CONSTEXPR __forceinline
-#elif defined(__GNUC__) && __GNUC__ >= 4
-#define BOOST_RANDOM_DETAIL_CONSTEXPR __attribute__((const)) __attribute__((always_inline))
+//#elif defined(__GNUC__) && __GNUC__ >= 4
+//#define BOOST_RANDOM_DETAIL_CONSTEXPR __attribute__((const)) __attribute__((always_inline))
 #else
 #define BOOST_RANDOM_DETAIL_CONSTEXPR inline
 #endif
diff -u -r boost_1_47_0/boost/uuid/seed_rng.hpp boost_1_47_0_tl/boost/uuid/seed_rng.hpp
--- boost_1_47_0/boost/uuid/seed_rng.hpp	2011-05-30 10:41:38.000000000 +1000
+++ boost_1_47_0_tl/boost/uuid/seed_rng.hpp	2011-08-09 09:53:45.000000000 +1000
@@ -139,7 +139,12 @@
         }
 
         {
-            unsigned int rn[] = { std::rand(), std::rand(), std::rand() };
+            unsigned int rn[] = 
+              { 
+                static_cast<unsigned int>(std::rand()), 
+                static_cast<unsigned int>(std::rand()), 
+                static_cast<unsigned int>(std::rand())
+              };
             sha.process_bytes( (unsigned char const*)rn, sizeof( rn ) );
         }
 
@@ -240,7 +245,10 @@
 {
     seed_rng seed_gen;
     generator_iterator<seed_rng> begin(&seed_gen);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wuninitialized"
     generator_iterator<seed_rng> end;
+#pragma GCC diagnostic pop
     rng.seed(begin, end);
 }
 
